Library/ViewModels/RewardsCollectionViewModel.swift

```diff
@@ -452,7 +478,27 @@ private func shouldTriggerEditRewardPrompt(_ data: PledgeViewData) -> Bool {
   return backing.addOns?.isEmpty == false && rewardChanged
 }
 
-private func backedReward(_ project: Project, rewards: [Reward]) -> IndexPath? {
+/// Returns the `IndexPath` of the reward to auto-scroll to in the collection view.
+/// If a `secretRewardToken` is provided, it returns the first secret reward's index.
+/// Otherwise, it returns the index of the backed reward (if the project is backed).
+private func rewardToScrollIndexPath(
+  _ project: Project,
+  rewards: [Reward],
+  secretRewardToken: String?
+) -> IndexPath? {
+  if let secretRewardToken = secretRewardToken, !secretRewardToken.isEmpty {
+    return firstSecretRewardIndexPath(rewards: rewards)
+  }
+
+  return backedRewardIndexPath(project, rewards: rewards)
+}
+
+private func firstSecretRewardIndexPath(rewards: [Reward]) -> IndexPath? {
+  return rewards.firstIndex(where: { $0.isSecretReward })
+    .flatMap { IndexPath(row: $0, section: 0) }
+}
```

Comment:
Are secret rewards provided to us in any particular order?
I'm wondering, if a project has multiple secret rewards, would we want to always scroll to the first one? Or would tokens be associated with specific secret rewards so that we could scroll to the most relevant one?